cmd/gateway-azure.go

```diff
@@ -550,10 +547,20 @@ func (a *azureObjects) PutObjectPart(bucket, object, uploadID string, partID int
 		teeReader = io.TeeReader(data, io.MultiWriter(writers...))
 	}
 
-	id := azureGetBlockID(partID, etag)
-	err = a.client.PutBlockWithLength(bucket, object, id, uint64(size), teeReader, nil)
-	if err != nil {
-		return info, azureToObjectError(traceError(err), bucket, object)
+	subPartSize := int64(azureBlockSize)
+	subPartNumber := 1
+	for remainingSize := size; remainingSize > 0; remainingSize -= subPartSize {
```

Comment:
this does not work for the case where we upload just one part with 0 size and do completemultipart of that 0 sized part

Reply:
Is zero sized part allowed in s3?

Reply:
yes

Reply:
This means gateway supposes to fail  for parts other than last part sizing < 5MB?

Reply:
Fixed now